crates/sui-core/src/execution_cache/writeback_cache.rs

```diff
@@ -2268,6 +2268,9 @@ impl StateSyncAPI for WritebackCache {
         transaction: &VerifiedTransaction,
         transaction_effects: &TransactionEffects,
     ) {
+        self.store
```

Comment:
Can the problem go away if we make this function atomic? (I forgot about some details and I seemed to recall that we discussed atomicity of the cache, and it was either complicate or not necessary before)

Reply:
Atomicity would fix it, but the cache intentionally eschews atomicity in order to simplify the transition to distributed storage in the future.